src/blurb/blurb.py

```diff
@@ -868,10 +876,115 @@ def find_editor():
     error('Could not find an editor! Set the EDITOR environment variable.')
 
 
+def _extract_issue_number(issue):
+    if issue is None:
+        return None
+
+    issue = raw_issue = str(issue).strip()
+    if issue.startswith('gh-'):
+        issue = issue[3:]
+    if issue.isdigit():
+        return issue
+
+    match = re.match(r'^(?:https://)?github\.com/python/cpython/issues/(\d+)$', issue)
+    if match is None:
+        sys.exit(f"Invalid GitHub issue: {raw_issue}")
+    return match.group(1)
+
+
+def _extract_section_name(section):
+    if section is None:
+        return None
+
+    section = raw_section = section.strip()
+    if section.strip('+-').isdigit():
+        section_index = int(section) - 1
+        if not (0 <= section_index < len(sections)):
+            sys.exit(f"Invalid section ID: {int(section)}\n\n"
+                     f"Choose from the following table:\n\n"
+                     f'{sections_table}')
+        return sections[section_index]
+
+    if not section:
+        sys.exit(f"Empty section name!")
+
+    sanitized = re.sub(r'[_-]', ' ', section)
+    section_words = re.split(r'\s+', sanitized)
+    section_pattern = '[_ ]'.join(map(re.escape, section_words))
+    section_re = re.compile(section_pattern, re.I)
+
+    matches = []
+    for section_name in sections:
+        if section_re.match(section_name):
+            matches.append(section_name)
+
+    if not matches:
+        sys.exit(f"Invalid section name: {raw_section}\n\n"
+                 f"Choose from the following table:\n\n"
+                 f'{sections_table}')
+
+    if len(matches) > 1:
+        multiple_matches = ', '.join(map(repr, sorted(matches)))
+        sys.exit(f"More than one match for: {raw_section}\n"
+                 f"Matches: {multiple_matches}\n\n"
+                 f"Choose from the following table:\n\n"
```

Comment:
If the user is sophisticated enough to use -s|--section with an abbreviation, it seems like it'd be sufficient to just show them the multiple matches.  I wouldn't show them the complete table of sections.  On the other hand, I would present the matching section names on individual lines.  Would you consider formatting it this way?

Reply:
Like, if there is an error -> show the table? sure it's something I can do!